### PR TITLE
Added support for multi-line comments

### PR DESCRIPTION
**Pull request by Dennis1543**
Support for multi-line comments: #79

### CODE REVIEW

**Comment by Dennis1543** [comment]
Example:
image

If there is no "\*/" found it defaults to commenting everything after "/\*".
image

File: Source/Parser/Expressions/ExpressionBase.cs [hunk 1]
```diff
@@ -122,6 +123,35 @@ internal static void SkipWhitespace(PositionalTokenizer tokenizer)
...
+                    });
+                }
+                else
+                {
+                    tokenizer.ReadTo("*/");
+                    tokenizer.Advance(2);
+                }
+
+                tokenizer.SkipWhitespace();
+            }
```
Comment (Jamiras) [inline Source/Parser/Expressions/ExpressionBase.cs:154]: This doesn't handle switching back and forth between single line comments and multiline comments:
```
// single
/* multi */
// single
/* multi */
code()
```

It matches 0 or more single, then 0 or more multi, then doesn't look for any more singles.

Comment (Jamiras) [reply Source/Parser/Expressions/ExpressionBase.cs:154]: Please add unit tests for this behavior. Here's some quick ones I wrote up the should be added near the existing ones in ExpressionBaseTests:
```
        [Test]
        public void TestSkipWhitespaceMultilineComment()
        {
            var tokenizer = new PositionalTokenizer(Tokenizer.CreateTokenizer("  /* multi-line\r\n     comment\r\n    */\r\nhi"));
            ExpressionBase.SkipWhitespace(tokenizer);
            Assert.That(tokenizer.NextChar, Is.EqualTo('h'));
        }

        [Test]
        public void TestSkipWhitespaceMultipleMultilineComment()
        {
            var tokenizer = new PositionalTokenizer(Tokenizer.CreateTokenizer("  /* multi-line\r\n     comment\r\n    */\r\n/* second */\r\nhi"));
            ExpressionBase.SkipWhitespace(tokenizer);
            Assert.That(tokenizer.NextChar, Is.EqualTo('h'));
        }

        [Test]
        public void TestSkipWhitespaceMultilineCommentUnclosed()
        {
            var tokenizer = new PositionalTokenizer(Tokenizer.CreateTokenizer("  /* multi-line\r\n     comment\r\nhi"));
            ExpressionBase.SkipWhitespace(tokenizer);
            Assert.That(tokenizer.NextChar, Is.EqualTo('\0'));
        }

        [Test]
        public void TestSkipWhitespaceMultilineMixedComment()
        {
            var tokenizer = new PositionalTokenizer(Tokenizer.CreateTokenizer("  // single-line\r\n/* multi-line\r\n     comment\r\n    */\r\n// second single line\r\n/* second multiline */\r\nhi"));
            ExpressionBase.SkipWhitespace(tokenizer);
            Assert.That(tokenizer.NextChar, Is.EqualTo('h'));
        }
```

**Review by Jamiras — CHANGES_REQUESTED** [review]

File: Source/Parser/Expressions/ExpressionBase.cs (hunk 1 above)
Comment (Dennis1543) [reply Source/Parser/Expressions/ExpressionBase.cs:158]: Switching back and forth between single and multiline comments should now be fixed. I also added these unit tests into ExpressionBaseTests.

**Review by Jamiras — APPROVED** [review]

**Pull request merged**